transport/src/androidTest/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImplTest.kt

```diff
@@ -1457,6 +1457,93 @@ class MessagingClientImplTest {
+        assertThat(subject.currentState).isIdle()
+        assertFailsWith<IllegalStateException>("MessagingClient is not in Configured or ReadOnly state.") { subject.clearConversation() }
+
+        subject.connect()
+        // currentState = Reconnecting
+        every { mockReconnectionHandler.shouldReconnect } returns true
+        val givenException = Exception(ErrorMessage.InternetConnectionIsOffline)
+        slot.captured.onFailure(givenException, ErrorCode.WebsocketError)
+
+        assertThat(subject.currentState).isReconnecting()
+        assertFailsWith<IllegalStateException>("MessagingClient is not in Configured or ReadOnly state.") { subject.clearConversation() }
+
+        // currentState = Closed
+        subject.disconnect()
+        assertThat(subject.currentState).isClosed(1000, "The user has closed the connection.")
+        assertFailsWith<IllegalStateException>("MessagingClient is not in Configured or ReadOnly state.") { subject.clearConversation() }
+
+        // currentState = Error
+        every { mockPlatformSocket.sendMessage(Request.configureRequest()) } answers {
+            slot.captured.onMessage(Response.sessionNotFound)
+        }
+        subject.connect()
+        assertThat(subject.currentState).isError(ErrorCode.SessionNotFound, "session not found error message")
+        assertFailsWith<IllegalStateException>("MessagingClient is not in Configured or ReadOnly state.") { subject.clearConversation() }
+    }
+
+    @Test
+    fun whenClearConversationIsDisabledInDeploymentConfigAndClearConversationIsCalled() {
+        every { mockDeploymentConfig.get() } returns createDeploymentConfigForTesting(
+            messenger = createMessengerVOForTesting(
+                apps = Apps(
+                    conversations = createConversationsVOForTesting(
+                        conversationClear = Conversations.ConversationClear(enabled = false)
+                    )
+                )
+            )
+        )
+        val expectedEvent = Event.Error(
+            errorCode = ErrorCode.ClearConversationFailure,
+            message = ErrorMessage.FailedToClearConversation,
+            correctiveAction = CorrectiveAction.Forbidden,
+        )
+        subject.connect()
+
+        subject.clearConversation()
+
+        verifySequence {
+            connectSequence()
+            mockEventHandler.onEvent(eq(expectedEvent))
+        }
+        verify(exactly = 0) {
+            mockPlatformSocket.sendMessage(eq(Request.clearConversation))
+        }
+    }
+
+    @Test
+    fun whenSessionIsConfiguredAndClearConversationCalled() {
+        subject.connect()
+
+        subject.clearConversation()
+
+        assertThat(subject.currentState).isConfigured(connected = true, newSession = true)
+        verifySequence {
+            connectSequence()
+            mockPlatformSocket.sendMessage(eq(Request.clearConversation))
+        }
+    }
+
+    @Test
+    fun whenSessionIsReadOnlyAndClearConversationCalled() {
```

Comment:
What is the point of clearing history of a readonly session. This make the whole session existence redundant. 
Won't it be better just to expire the session or something?

Reply:
No matter state(as soon as it active) for privacy reasons, user must have capability to clear the conversation history.
As part of the flow SessionClearedEvent will be dispatched by Shyrka following by ConnectionClosedEvent. 
At ConnectionClosedEvent Transport will close the ws connection, forcing UI to establish new connection. 
N

Reply:
No matter state(as soon as it is active) user must have capability to clear the conversation history. This is done for privacy reasons.
Actually, once clearconversation is called some sort of session expiry does happen. As part of the flow SessionClearedEvent will be dispatched by Shyrka following by ConnectionClosedEvent.
At ConnectionClosedEvent Transport will close the ws connection, forcing UI to establish new one.
So if you do that from ReadOnly state the new configureSession request will result in new session in state Configured with all what it implies(like Autorsart request and etc).